src/Database/Esqueleto/PostgreSQL/JSON.hs

```diff
@@ -0,0 +1,618 @@
+--
+-- === __Objects__
+--
+-- @
+-- {"a": 3.14, "b": null}        #- []        == {"a": 3.14, "b": null}
+-- {"a": 3.14, "b": null}        #- ["a"]     == {"b": null}
+-- {"a": 3.14, "b": null}        #- ["a","b"] == {"a": 3.14, "b": null}
+-- {"a": {"b":false}, "b": null} #- ["a","b"] == {"a": {}, "b": null}
+-- @
+--
+-- === __Arrays__
+--
+-- [true, {"b":null}, 5]       #- []            == [true, {"b":null}, 5]
+-- [true, {"b":null}, 5]       #- ["0"]         == [{"b":null}, 5]
+-- [true, {"b":null}, 5]       #- ["b"]         == ERROR: path element at position 1 is not an integer: "b"
+-- [true, {"b":null}, 5]       #- ["1","b"]     == [true, {}, 5]
+-- [true, {"b":null}, 5]       #- ["-2","b"]    == [true, {}, 5]
+-- {"a": {"b":[false,4,null]}} #- ["a","b","2"] == {"a": {"b":[false,4]}}
+-- {"a": {"b":[false,4,null]}} #- ["a","b","c"] == ERROR: path element at position 3 is not an integer: "c"
+--
+-- === __Other values__
+--
+-- 1    #- {anything} == ERROR: cannot delete from scalar
+-- "a"  #- {anything} == ERROR: cannot delete from scalar
+-- true #- {anything} == ERROR: cannot delete from scalar
+-- null #- {anything} == ERROR: cannot delete from scalar
+-- @
+--
+-- === __PostgreSQL Documentation__
+--
+-- /Works with any PostgreSQL of version >= 9.5/
+--
+-- @
+--     | Type   | Description                                             |  Example
+-- ----+--------+---------------------------------------------------------+------------------------------------
+--  #- | text[] | Delete the field or element with specified path         | '["a", {"b":1}]'::jsonb #- '{1,b}'
+--     |        | (for JSON arrays, negative integers count from the end) |
+-- @
+--
+-- @since 3.1.0
+(#-.) :: SqlExpr (Value (Maybe Aeson.Value))
+      -> [Text]
+      -> SqlExpr (Value (Maybe Aeson.Value))
+(#-.) value = unsafeSqlBinOp " #- " value . mkTextArray
+
+mkTextArray :: [Text] -> SqlExpr (Value PersistValue)
+mkTextArray = val . PersistArray . fmap toPersistValue
+
+
+
+--- ORPHAN INSTANCES ---
+--- ORPHAN INSTANCES ---
+--- ORPHAN INSTANCES ---
+
+
+-- Mainly copied over from Database.Persist.Postgresql.JSON
+-- Since we don't need anything else and adding another dependency
+-- just for these two instances is a bit overkill.
+
+-- | @since 3.1.0
+instance PersistField Aeson.Value where
+  toPersistValue = PersistDbSpecific . BSL.toStrict . encode
+  fromPersistValue pVal = case pVal of
+      PersistByteString bs -> fromLeft (badParse $ TE.decodeUtf8 bs) $ eitherDecodeStrict bs
+      PersistText t -> fromLeft (badParse t) $ eitherDecodeStrict (TE.encodeUtf8 t)
+      x -> Left $ fromPersistValueError "string or bytea" x
+
+-- | jsonb - @since 3.1.0
+instance PersistFieldSql Aeson.Value where
+  sqlType _ = SqlOther "JSONB"
+
+badParse :: Text -> String -> Text
+badParse t = fromPersistValueParseError t . T.pack
+
+fromLeft :: (a -> b) -> Either a x -> Either b x
+fromLeft f (Left a) = Left $ f a
+fromLeft _ (Right r) = Right r
```

Comment:
This is in `base` via `Data.Bifunctor` as `first`

Reply:
Ah, you're right P: